blackbird_python/blackbird/tests/test_program.py

```diff
@@ -109,14 +111,24 @@ def test_invalid_template_call(self):
         bb = BlackbirdProgram(name="prog", version=0.0)
         x = sym.Symbol('x')
         hi = sym.Symbol('hi')
-        bb._parameters = [x, hi]
+        y = sym.Symbol('y')
+        bye = sym.Symbol('bye')
+
+        bb._parameters = [x, hi, y, bye]
         bb._operations.append({"op": "Dgate", "modes": [0], "args": [0.54/x**2], "kwargs": {'test': hi**4}})
+        bb._var.update({"y": y, "bye": np.array([[1, 2, bye]])})
```

Comment:
Nice!

Although, it might make sense to have a separate test for array parameters, since the logic is quite different